pyquil/api/_qam.py

```diff
@@ -60,18 +61,27 @@ def load(self, executable):
         return self
 
     @_record_call
-    def write_memory(self, *, region_name: str, offset: int = 0, value=None):
+    def write_memory(self, *, region_name: str, offset: int = None,
+                     value: Optional[Union[int, float, Sequence[int], Sequence[float]]] = None):
         """
-        Writes a value into a memory region on the QAM at a specified offset.
+        Writes a value or chronologically unwraps a list of values into a memory region on
+        the QAM at a specified offset.
 
         :param region_name: Name of the declared memory region on the QAM.
         :param offset: Integer offset into the memory region to write to.
-        :param value: Value to store at the indicated location.
+        :param value: Value(s) to store at the indicated location.
         """
         assert self.status in ['loaded', 'done']
 
-        aref = ParameterAref(name=region_name, index=offset)
-        self._variables_shim[aref] = value
+        if isinstance(value, Sequence):
+            if offset != None:
+                warnings.warn("offset should be None when value is a Sequence")
+            for index, v in enumerate(value):
+                aref = ParameterAref(name=region_name, index=offset + index)
```

Comment:
Tests are failing here when `offset` is `None`. See my comment in the previous PR here: https://github.com/rigetti/pyquil/pull/1079#discussion_r349235805